src/dom/unload.ts

```diff
@@ -1,9 +1,19 @@
-// @ifdef DEBUG
-window.onbeforeunload = function(e) {
-    return undefined;
+const MSG = "This page navigation attempt is likely to be a popunder (with popup blocked) do you wish to continue?";
+
+const onbeforeunloadHandler = (evt) => {
+    evt.returnValue = MSG;
+    return MSG;
 };
 
-window.addEventListener('unload', function() {
-    
-});
-// @endif
+export const setBeforeunloadHandler = () => {
+    // ToDo: if this found to be useful, consider making it work on cross-origin iframes
+    if (window === window.top) {
+        const prev = window.onbeforeunload;
+        window.onbeforeunload = onbeforeunloadHandler;
+        setTimeout(() => {
+            if (onbeforeunloadHandler === window.onbeforeunload) {
```

Comment:
It doesn't provide much practical value, it can be still overridden by `window.addEventListener('beforeunload', someOtherHandler);`
If we override `EventTarget.prototype.addEventListener`, we can prevent this as well, but I'm not sure if we should override another common api just for this purpose.